<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Kaixuan Luo" />

<meta name="date" content="2025-06-03" />

<title>Aggregate profiles around motif sites</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">footprint_clustering</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Aggregate profiles around motif sites</h1>
<h4 class="author">Kaixuan Luo</h4>
<h4 class="date">2025-06-03</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-06-03
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>footprint_clustering/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250530code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250530)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250530code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250530)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkevinlkxfootprintclusteringtree8ab13cbac998a0316815e02ee34f852fdbae935atargetblank8ab13cba">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/kevinlkx/footprint_clustering/tree/8ab13cbac998a0316815e02ee34f852fdbae935a" target="_blank">8ab13cb</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomkevinlkxfootprintclusteringtree8ab13cbac998a0316815e02ee34f852fdbae935atargetblank8ab13cba"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/kevinlkx/footprint_clustering/tree/8ab13cbac998a0316815e02ee34f852fdbae935a" target="_blank">8ab13cb</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/plot_aggregate_footprint_profiles.Rmd</code>) and HTML
(<code>docs/plot_aggregate_footprint_profiles.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kevinlkx/footprint_clustering/blob/8ab13cbac998a0316815e02ee34f852fdbae935a/analysis/plot_aggregate_footprint_profiles.Rmd" target="_blank">8ab13cb</a>
</td>
<td>
kevinlkx
</td>
<td>
2025-06-03
</td>
<td>
wflow_publish("analysis/plot_aggregate_footprint_profiles.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/kevinlkx/footprint_clustering/a7416a5759dbd544ba23002b4578d7a0f513c651/docs/plot_aggregate_footprint_profiles.html" target="_blank">a7416a5</a>
</td>
<td>
kevinlkx
</td>
<td>
2025-06-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kevinlkx/footprint_clustering/blob/b689be296e5661d5c57073faec37f9d51cbdc4ff/analysis/plot_aggregate_footprint_profiles.Rmd" target="_blank">b689be2</a>
</td>
<td>
kevinlkx
</td>
<td>
2025-06-03
</td>
<td>
wflow_publish("analysis/plot_aggregate_footprint_profiles.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/kevinlkx/footprint_clustering/b3cb0f0cba10ad3555a6c15fb30d78880619489d/docs/plot_aggregate_footprint_profiles.html" target="_blank">b3cb0f0</a>
</td>
<td>
kevinlkx
</td>
<td>
2025-06-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kevinlkx/footprint_clustering/blob/0f500a08c0ad187fbf989be07e33ba441820b812/analysis/plot_aggregate_footprint_profiles.Rmd" target="_blank">0f500a0</a>
</td>
<td>
kevinlkx
</td>
<td>
2025-06-03
</td>
<td>
wflow_publish("analysis/plot_aggregate_footprint_profiles.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/kevinlkx/footprint_clustering/2c0e42e87988eb6a6d5acaecb20a9f8176f327f7/docs/plot_aggregate_footprint_profiles.html" target="_blank">2c0e42e</a>
</td>
<td>
kevinlkx
</td>
<td>
2025-06-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kevinlkx/footprint_clustering/blob/be43eb73da8b77b1aa996ef813f2e08cfad8aeff/analysis/plot_aggregate_footprint_profiles.Rmd" target="_blank">be43eb7</a>
</td>
<td>
kevinlkx
</td>
<td>
2025-06-03
</td>
<td>
wflow_publish("analysis/plot_aggregate_footprint_profiles.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(data.table)
library(ggplot2)
source(&quot;code/plots.R&quot;)</code></pre>
<p>Here, we explore the accessibility profiles round TF motif match
sites.</p>
<div id="ctcf" class="section level2">
<h2>CTCF</h2>
<div id="ctcf-profiles-in-gm12878" class="section level3">
<h3>CTCF profiles in GM12878</h3>
<p>We scanned CTCF motif matches along the genome, then extended 100bp
window on both sides of motif matches.</p>
<p>We downloaded DNase-seq, ATAC-seq and CTCF ChIP-seq data in GM12878
from ENCODE.</p>
<p>About the processed files:</p>
<ul>
<li><p><code>CTCF_MA0139.2_1e-5.candidate.sites.rds</code>: a data frame
of candidate binding sites matching CTCF motif (MA0139.2) (result from
FIMO, p-value &lt; 1e-5).</p></li>
<li><p><code>CTCF.GM12878.sites.chip.labels.rds</code>: a data frame
with the CTCF motif matches (result from FIMO, as in
<code>*.candidate.sites.rds</code>), as well as normalized ChIP-seq
counts (“chip” column) and ChIP-seq peak labels (“chip_label”) in
GM12878.</p></li>
<li><p><code>*.DNase.counts.mat.rds</code>: a matrix where the rows are
the motif matches (in the same order as <code>sites</code>), and the
columns are the DNase-seq counts at each position in the window. The
first half of the columns are counts on the forward strand, and the
second half are counts on the reverse strand. We could simple combine
the counts on both strands as shown below.</p></li>
<li><p><code>*.ATAC.counts.mat.rds</code>: ATAC-seq count matrix,
similar to <code>*.DNase.counts.mat.rds</code>.</p></li>
</ul>
<pre class="r"><code>sites &lt;- readRDS(&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/CTCF_MA0139.2_1e-5.candidate.sites.rds&#39;)
sites_chip_labels &lt;- readRDS(&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/CTCF.GM12878.sites.chip.labels.rds&#39;)

dnase_count_matrix &lt;- readRDS(&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/CTCF.GM12878.DNase.counts.mat.rds&#39;)
# combine counts on both strands
dnase_count_matrix &lt;- dnase_count_matrix[,1:(ncol(dnase_count_matrix)/2)] + dnase_count_matrix[,(ncol(dnase_count_matrix)/2+1):ncol(dnase_count_matrix)]

atac_count_matrix &lt;- readRDS(&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/CTCF.GM12878.ATAC.counts.mat.rds&#39;)
# combine counts on both strands
atac_count_matrix &lt;- atac_count_matrix[,1:(ncol(atac_count_matrix)/2)] + atac_count_matrix[,(ncol(atac_count_matrix)/2+1):ncol(atac_count_matrix)]</code></pre>
<p>Plot aggregate profiles of DNase-seq and ATAC-seq counts around motif
sites.</p>
<pre class="r"><code>pos_idx &lt;- which(sites_chip_labels$chip_label == 1)
neg_idx &lt;- which(sites_chip_labels$chip_label == 0)

dnase_pos_profile &lt;- colMeans(dnase_count_matrix[pos_idx, ], na.rm = TRUE)
dnase_neg_profile &lt;- colMeans(dnase_count_matrix[neg_idx, ], na.rm = TRUE)
plot_pos_neg_profiles(dnase_pos_profile, dnase_neg_profile, 
                      title = &quot;Aggregate DNase-seq profiles around CTCF motifs in GM12878&quot;)</code></pre>
<p><img src="figure/plot_aggregate_footprint_profiles.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kevinlkx/footprint_clustering/blob/2c0e42e87988eb6a6d5acaecb20a9f8176f327f7/docs/figure/plot_aggregate_footprint_profiles.Rmd/unnamed-chunk-3-1.png" target="_blank">2c0e42e</a>
</td>
<td>
kevinlkx
</td>
<td>
2025-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>atac_pos_profile &lt;- colMeans(atac_count_matrix[pos_idx, ], na.rm = TRUE)
atac_neg_profile &lt;- colMeans(atac_count_matrix[neg_idx, ], na.rm = TRUE)
plot_pos_neg_profiles(atac_pos_profile, atac_neg_profile, 
                      title = &quot;Aggregate ATAC-seq profiles around CTCF motifs in GM12878&quot;)</code></pre>
<p><img src="figure/plot_aggregate_footprint_profiles.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-2">
Past versions of unnamed-chunk-3-2.png
</button>
</p>
<div id="fig-unnamed-chunk-3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kevinlkx/footprint_clustering/blob/2c0e42e87988eb6a6d5acaecb20a9f8176f327f7/docs/figure/plot_aggregate_footprint_profiles.Rmd/unnamed-chunk-3-2.png" target="_blank">2c0e42e</a>
</td>
<td>
kevinlkx
</td>
<td>
2025-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Heatmap of 2000 positive sites and 2000 negative sites</p>
<p>We randomly sample 2000 positive sites and 2000 negative sites, and
plot the PWM scores, DNase-seq, ATAC-seq and ChIP-seq results.</p>
<pre class="r"><code>set.seed(1)
sites_idx &lt;- c(sample(pos_idx, 2000), sample(neg_idx, 2000))

pwm = sites_chip_labels$pwm.score[sites_idx]
chip = sites_chip_labels$chip[sites_idx]
chip_label = sites_chip_labels$chip_label[sites_idx]
dnase_data = dnase_count_matrix[sites_idx,]
atac_data = atac_count_matrix[sites_idx,]

rank = order(chip_label, chip)

data.l &lt;- list(DNaase = dnase_data,
               ATAC = atac_data)
chip.df &lt;- data.frame(chip = chip, chip_label = chip_label)

plot_data_matrix_heatmap(pwm, data.l, chip.df, rank, 
                         data_name = c(&quot;DNase-seq&quot;, &quot;ATAC-seq&quot;), 
                         chip_name = c(&quot;ChIP-seq counts&quot;, &quot;ChIP-seq peaks&quot;), 
                         title = &quot;CTCF sites in GM12878&quot;,
                         zMax_data = c(1, 3), 
                         zMax_chip = c(200, 1))</code></pre>
<p><img src="figure/plot_aggregate_footprint_profiles.Rmd/unnamed-chunk-4-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kevinlkx/footprint_clustering/blob/2c0e42e87988eb6a6d5acaecb20a9f8176f327f7/docs/figure/plot_aggregate_footprint_profiles.Rmd/unnamed-chunk-4-1.png" target="_blank">2c0e42e</a>
</td>
<td>
kevinlkx
</td>
<td>
2025-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="ctcf-profiles-in-k562" class="section level3">
<h3>CTCF profiles in K562</h3>
<p>We scanned CTCF motif matches along the genome, then extended 100bp
window on both sides of motif matches.</p>
<p>We downloaded DNase-seq, ATAC-seq and CTCF ChIP-seq data in K562 from
ENCODE.</p>
<pre class="r"><code># sites &lt;- readRDS(&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/CTCF_MA0139.2_1e-5.candidate.sites.rds&#39;)

sites_chip_labels &lt;- readRDS(&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/CTCF.K562.sites.chip.labels.rds&#39;)

dnase_count_matrix &lt;- readRDS(&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/CTCF.K562.DNase.counts.mat.rds&#39;)
# combine counts on both strands
dnase_count_matrix &lt;- dnase_count_matrix[,1:(ncol(dnase_count_matrix)/2)] + dnase_count_matrix[,(ncol(dnase_count_matrix)/2+1):ncol(dnase_count_matrix)]

atac_count_matrix &lt;- readRDS(&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/CTCF.K562.ATAC.counts.mat.rds&#39;)
# combine counts on both strands
atac_count_matrix &lt;- atac_count_matrix[,1:(ncol(atac_count_matrix)/2)] + atac_count_matrix[,(ncol(atac_count_matrix)/2+1):ncol(atac_count_matrix)]</code></pre>
<p>Plot aggregate profiles of DNase-seq and ATAC-seq counts around motif
sites.</p>
<pre class="r"><code>pos_idx &lt;- which(sites_chip_labels$chip_label == 1)
neg_idx &lt;- which(sites_chip_labels$chip_label == 0)

dnase_pos_profile &lt;- colMeans(dnase_count_matrix[pos_idx, ], na.rm = TRUE)
dnase_neg_profile &lt;- colMeans(dnase_count_matrix[neg_idx, ], na.rm = TRUE)
plot_pos_neg_profiles(dnase_pos_profile, dnase_neg_profile, 
                      title = &quot;Aggregate DNase-seq profiles around CTCF motifs in K562&quot;)</code></pre>
<p><img src="figure/plot_aggregate_footprint_profiles.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kevinlkx/footprint_clustering/blob/2c0e42e87988eb6a6d5acaecb20a9f8176f327f7/docs/figure/plot_aggregate_footprint_profiles.Rmd/unnamed-chunk-6-1.png" target="_blank">2c0e42e</a>
</td>
<td>
kevinlkx
</td>
<td>
2025-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>atac_pos_profile &lt;- colMeans(atac_count_matrix[pos_idx, ], na.rm = TRUE)
atac_neg_profile &lt;- colMeans(atac_count_matrix[neg_idx, ], na.rm = TRUE)
plot_pos_neg_profiles(atac_pos_profile, atac_neg_profile, 
                      title = &quot;Aggregate ATAC-seq profiles around CTCF motifs in K562&quot;)</code></pre>
<p><img src="figure/plot_aggregate_footprint_profiles.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-2">
Past versions of unnamed-chunk-6-2.png
</button>
</p>
<div id="fig-unnamed-chunk-6-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kevinlkx/footprint_clustering/blob/2c0e42e87988eb6a6d5acaecb20a9f8176f327f7/docs/figure/plot_aggregate_footprint_profiles.Rmd/unnamed-chunk-6-2.png" target="_blank">2c0e42e</a>
</td>
<td>
kevinlkx
</td>
<td>
2025-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Heatmap of 2000 positive sites and 2000 negative sites</p>
<p>We randomly sample 2000 positive sites and 2000 negative sites, and
plot the PWM scores, DNase-seq, ATAC-seq and ChIP-seq results.</p>
<pre class="r"><code>set.seed(1)
sites_idx &lt;- c(sample(pos_idx, 2000), sample(neg_idx, 2000))

pwm = sites_chip_labels$pwm.score[sites_idx]
chip = sites_chip_labels$chip[sites_idx]
chip_label = sites_chip_labels$chip_label[sites_idx]
dnase_data = dnase_count_matrix[sites_idx,]
atac_data = atac_count_matrix[sites_idx,]

rank = order(chip_label, chip)

data.l &lt;- list(DNaase = dnase_data,
               ATAC = atac_data)
chip.df &lt;- data.frame(chip = chip, chip_label = chip_label)

plot_data_matrix_heatmap(pwm, data.l, chip.df, rank, 
                         data_name = c(&quot;DNase-seq&quot;, &quot;ATAC-seq&quot;), 
                         chip_name = c(&quot;ChIP-seq counts&quot;, &quot;ChIP-seq peaks&quot;), 
                         title = &quot;CTCF sites in K562&quot;,
                         zMax_data = c(1, 3), 
                         zMax_chip = c(200, 1))</code></pre>
<p><img src="figure/plot_aggregate_footprint_profiles.Rmd/unnamed-chunk-7-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kevinlkx/footprint_clustering/blob/2c0e42e87988eb6a6d5acaecb20a9f8176f327f7/docs/figure/plot_aggregate_footprint_profiles.Rmd/unnamed-chunk-7-1.png" target="_blank">2c0e42e</a>
</td>
<td>
kevinlkx
</td>
<td>
2025-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="rest" class="section level2">
<h2>REST</h2>
<div id="rest-profiles-in-gm12878" class="section level3">
<h3>REST profiles in GM12878</h3>
<p>We scanned REST motif matches along the genome, then extended 100bp
window on both sides of motif matches.</p>
<p>We downloaded DNase-seq, ATAC-seq and REST ChIP-seq data in GM12878
from ENCODE.</p>
<pre class="r"><code># sites &lt;- readRDS(&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/REST_MA0138.3_1e-5.candidate.sites.rds&#39;)

sites_chip_labels &lt;- readRDS(&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/REST.GM12878.sites.chip.labels.rds&#39;)

dnase_count_matrix &lt;- readRDS(&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/REST.GM12878.DNase.counts.mat.rds&#39;)
# combine counts on both strands
dnase_count_matrix &lt;- dnase_count_matrix[,1:(ncol(dnase_count_matrix)/2)] + dnase_count_matrix[,(ncol(dnase_count_matrix)/2+1):ncol(dnase_count_matrix)]

atac_count_matrix &lt;- readRDS(&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/REST.GM12878.ATAC.counts.mat.rds&#39;)
# combine counts on both strands
atac_count_matrix &lt;- atac_count_matrix[,1:(ncol(atac_count_matrix)/2)] + atac_count_matrix[,(ncol(atac_count_matrix)/2+1):ncol(atac_count_matrix)]</code></pre>
<p>Plot aggregate profiles of DNase-seq and ATAC-seq counts around motif
sites.</p>
<pre class="r"><code>pos_idx &lt;- which(sites_chip_labels$chip_label == 1)
neg_idx &lt;- which(sites_chip_labels$chip_label == 0)

dnase_pos_profile &lt;- colMeans(dnase_count_matrix[pos_idx, ], na.rm = TRUE)
dnase_neg_profile &lt;- colMeans(dnase_count_matrix[neg_idx, ], na.rm = TRUE)
plot_pos_neg_profiles(dnase_pos_profile, dnase_neg_profile, 
                      title = &quot;Aggregate DNase-seq profiles around REST motifs in GM12878&quot;)</code></pre>
<p><img src="figure/plot_aggregate_footprint_profiles.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kevinlkx/footprint_clustering/blob/2c0e42e87988eb6a6d5acaecb20a9f8176f327f7/docs/figure/plot_aggregate_footprint_profiles.Rmd/unnamed-chunk-9-1.png" target="_blank">2c0e42e</a>
</td>
<td>
kevinlkx
</td>
<td>
2025-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>atac_pos_profile &lt;- colMeans(atac_count_matrix[pos_idx, ], na.rm = TRUE)
atac_neg_profile &lt;- colMeans(atac_count_matrix[neg_idx, ], na.rm = TRUE)
plot_pos_neg_profiles(atac_pos_profile, atac_neg_profile, 
                      title = &quot;Aggregate ATAC-seq profiles around REST motifs in GM12878&quot;)</code></pre>
<p><img src="figure/plot_aggregate_footprint_profiles.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-2">
Past versions of unnamed-chunk-9-2.png
</button>
</p>
<div id="fig-unnamed-chunk-9-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kevinlkx/footprint_clustering/blob/2c0e42e87988eb6a6d5acaecb20a9f8176f327f7/docs/figure/plot_aggregate_footprint_profiles.Rmd/unnamed-chunk-9-2.png" target="_blank">2c0e42e</a>
</td>
<td>
kevinlkx
</td>
<td>
2025-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Heatmap of 2000 positive sites and 2000 negative sites</p>
<p>We randomly sample 2000 positive sites and 2000 negative sites, and
plot the PWM scores, DNase-seq, ATAC-seq and ChIP-seq results.</p>
<pre class="r"><code>set.seed(1)
sites_idx &lt;- c(sample(pos_idx, 2000), sample(neg_idx, 2000))

pwm = sites_chip_labels$pwm.score[sites_idx]
chip = sites_chip_labels$chip[sites_idx]
chip_label = sites_chip_labels$chip_label[sites_idx]
dnase_data = dnase_count_matrix[sites_idx,]
atac_data = atac_count_matrix[sites_idx,]

rank = order(chip_label, chip)

data.l &lt;- list(DNaase = dnase_data,
               ATAC = atac_data)
chip.df &lt;- data.frame(chip = chip, chip_label = chip_label)

plot_data_matrix_heatmap(pwm, data.l, chip.df, rank, 
                         data_name = c(&quot;DNase-seq&quot;, &quot;ATAC-seq&quot;), 
                         chip_name = c(&quot;ChIP-seq counts&quot;, &quot;ChIP-seq peaks&quot;), 
                         title = &quot;REST sites in GM12878&quot;,
                         zMax_data = c(1, 3), 
                         zMax_chip = c(200, 1))</code></pre>
<p><img src="figure/plot_aggregate_footprint_profiles.Rmd/unnamed-chunk-10-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kevinlkx/footprint_clustering/blob/2c0e42e87988eb6a6d5acaecb20a9f8176f327f7/docs/figure/plot_aggregate_footprint_profiles.Rmd/unnamed-chunk-10-1.png" target="_blank">2c0e42e</a>
</td>
<td>
kevinlkx
</td>
<td>
2025-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="ctcf-profiles-in-k562-1" class="section level3">
<h3>CTCF profiles in K562</h3>
<p>We scanned REST motif matches along the genome, then extended 100bp
window on both sides of motif matches.</p>
<p>We downloaded DNase-seq, ATAC-seq and REST ChIP-seq data in K562 from
ENCODE.</p>
<pre class="r"><code># sites &lt;- readRDS(&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/REST_MA0138.3_1e-5.candidate.sites.rds&#39;)

sites_chip_labels &lt;- readRDS(&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/REST.K562.sites.chip.labels.rds&#39;)

dnase_count_matrix &lt;- readRDS(&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/REST.K562.DNase.counts.mat.rds&#39;)
# combine counts on both strands
dnase_count_matrix &lt;- dnase_count_matrix[,1:(ncol(dnase_count_matrix)/2)] + dnase_count_matrix[,(ncol(dnase_count_matrix)/2+1):ncol(dnase_count_matrix)]

atac_count_matrix &lt;- readRDS(&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/REST.K562.ATAC.counts.mat.rds&#39;)
# combine counts on both strands
atac_count_matrix &lt;- atac_count_matrix[,1:(ncol(atac_count_matrix)/2)] + atac_count_matrix[,(ncol(atac_count_matrix)/2+1):ncol(atac_count_matrix)]</code></pre>
<p>Plot aggregate profiles of DNase-seq and ATAC-seq counts around motif
sites.</p>
<pre class="r"><code>pos_idx &lt;- which(sites_chip_labels$chip_label == 1)
neg_idx &lt;- which(sites_chip_labels$chip_label == 0)

dnase_pos_profile &lt;- colMeans(dnase_count_matrix[pos_idx, ], na.rm = TRUE)
dnase_neg_profile &lt;- colMeans(dnase_count_matrix[neg_idx, ], na.rm = TRUE)
plot_pos_neg_profiles(dnase_pos_profile, dnase_neg_profile, 
                      title = &quot;Aggregate DNase-seq profiles around REST motifs in K562&quot;)</code></pre>
<p><img src="figure/plot_aggregate_footprint_profiles.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kevinlkx/footprint_clustering/blob/2c0e42e87988eb6a6d5acaecb20a9f8176f327f7/docs/figure/plot_aggregate_footprint_profiles.Rmd/unnamed-chunk-12-1.png" target="_blank">2c0e42e</a>
</td>
<td>
kevinlkx
</td>
<td>
2025-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>atac_pos_profile &lt;- colMeans(atac_count_matrix[pos_idx, ], na.rm = TRUE)
atac_neg_profile &lt;- colMeans(atac_count_matrix[neg_idx, ], na.rm = TRUE)
plot_pos_neg_profiles(atac_pos_profile, atac_neg_profile, 
                      title = &quot;Aggregate ATAC-seq profiles around REST motifs in K562&quot;)</code></pre>
<p><img src="figure/plot_aggregate_footprint_profiles.Rmd/unnamed-chunk-12-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-2">
Past versions of unnamed-chunk-12-2.png
</button>
</p>
<div id="fig-unnamed-chunk-12-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kevinlkx/footprint_clustering/blob/2c0e42e87988eb6a6d5acaecb20a9f8176f327f7/docs/figure/plot_aggregate_footprint_profiles.Rmd/unnamed-chunk-12-2.png" target="_blank">2c0e42e</a>
</td>
<td>
kevinlkx
</td>
<td>
2025-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Heatmap of 2000 positive sites and 2000 negative sites</p>
<p>We randomly sample 2000 positive sites and 2000 negative sites, and
plot the PWM scores, DNase-seq, ATAC-seq and ChIP-seq results.</p>
<pre class="r"><code>set.seed(1)
sites_idx &lt;- c(sample(pos_idx, 2000), sample(neg_idx, 2000))

pwm = sites_chip_labels$pwm.score[sites_idx]
chip = sites_chip_labels$chip[sites_idx]
chip_label = sites_chip_labels$chip_label[sites_idx]
dnase_data = dnase_count_matrix[sites_idx,]
atac_data = atac_count_matrix[sites_idx,]

rank = order(chip_label, chip)

data.l &lt;- list(DNaase = dnase_data,
               ATAC = atac_data)
chip.df &lt;- data.frame(chip = chip, chip_label = chip_label)

plot_data_matrix_heatmap(pwm, data.l, chip.df, rank, 
                         data_name = c(&quot;DNase-seq&quot;, &quot;ATAC-seq&quot;), 
                         chip_name = c(&quot;ChIP-seq counts&quot;, &quot;ChIP-seq peaks&quot;), 
                         title = &quot;REST sites in K562&quot;,
                         zMax_data = c(1, 3), 
                         zMax_chip = c(200, 1))</code></pre>
<p><img src="figure/plot_aggregate_footprint_profiles.Rmd/unnamed-chunk-13-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/kevinlkx/footprint_clustering/blob/2c0e42e87988eb6a6d5acaecb20a9f8176f327f7/docs/figure/plot_aggregate_footprint_profiles.Rmd/unnamed-chunk-13-1.png" target="_blank">2c0e42e</a>
</td>
<td>
kevinlkx
</td>
<td>
2025-06-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>#&gt; R version 4.2.0 (2022-04-22)
#&gt; Platform: x86_64-pc-linux-gnu (64-bit)
#&gt; Running under: CentOS Linux 7 (Core)
#&gt; 
#&gt; Matrix products: default
#&gt; BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so
#&gt; 
#&gt; locale:
#&gt;  [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
#&gt;  [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
#&gt;  [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
#&gt; [10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 
#&gt; 
#&gt; attached base packages:
#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     
#&gt; 
#&gt; other attached packages:
#&gt;  [1] data.table_1.14.2 forcats_0.5.1     stringr_1.5.1     dplyr_1.1.4      
#&gt;  [5] purrr_1.0.2       readr_2.1.2       tidyr_1.3.1       tibble_3.2.1     
#&gt;  [9] ggplot2_3.5.1     tidyverse_1.3.1   workflowr_1.7.0  
#&gt; 
#&gt; loaded via a namespace (and not attached):
#&gt;  [1] tidyselect_1.2.1 xfun_0.30        bslib_0.3.1      haven_2.5.0     
#&gt;  [5] colorspace_2.0-3 vctrs_0.6.5      generics_0.1.2   htmltools_0.5.2 
#&gt;  [9] yaml_2.3.5       rlang_1.1.4      jquerylib_0.1.4  later_1.3.0     
#&gt; [13] pillar_1.10.1    withr_2.5.0      glue_1.6.2       DBI_1.1.2       
#&gt; [17] dbplyr_2.1.1     readxl_1.4.0     modelr_0.1.8     lifecycle_1.0.4 
#&gt; [21] cellranger_1.1.0 munsell_0.5.0    gtable_0.3.0     rvest_1.0.2     
#&gt; [25] evaluate_0.15    knitr_1.39       tzdb_0.3.0       callr_3.7.3     
#&gt; [29] fastmap_1.1.0    httpuv_1.6.5     ps_1.7.0         highr_0.9       
#&gt; [33] broom_0.8.0      Rcpp_1.0.12      promises_1.2.0.1 backports_1.4.1 
#&gt; [37] scales_1.3.0     jsonlite_1.8.0   fs_1.5.2         hms_1.1.1       
#&gt; [41] digest_0.6.29    stringi_1.7.6    processx_3.8.0   getPass_0.2-2   
#&gt; [45] rprojroot_2.0.3  grid_4.2.0       cli_3.6.3        tools_4.2.0     
#&gt; [49] magrittr_2.0.3   sass_0.4.1       crayon_1.5.1     whisker_0.4     
#&gt; [53] pkgconfig_2.0.3  ellipsis_0.3.2   xml2_1.3.3       reprex_2.0.1    
#&gt; [57] lubridate_1.8.0  assertthat_0.2.1 rmarkdown_2.14   httr_1.4.3      
#&gt; [61] rstudioapi_0.13  R6_2.5.1         git2r_0.30.1     compiler_4.2.0</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
