<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Kaixuan Luo" />

<meta name="date" content="2025-09-10" />

<title>Prepare input data in GM12878 cell line</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">footprint_clustering</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Prepare input data in GM12878 cell
line</h1>
<h4 class="author">Kaixuan Luo</h4>
<h4 class="date">2025-09-10</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-09-10
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>footprint_clustering/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250530code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250530)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250530code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250530)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkevinlkxfootprintclusteringtree8e46a4664959487556b298cc0e8f7894c086c1cftargetblank8e46a46a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/kevinlkx/footprint_clustering/tree/8e46a4664959487556b298cc0e8f7894c086c1cf" target="_blank">8e46a46</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomkevinlkxfootprintclusteringtree8e46a4664959487556b298cc0e8f7894c086c1cftargetblank8e46a46a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/kevinlkx/footprint_clustering/tree/8e46a4664959487556b298cc0e8f7894c086c1cf" target="_blank">8e46a46</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Unstaged changes:
    Modified:   analysis/prepare_input_data_K562.Rmd
    Modified:   code/get_motif_sites.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/prepare_input_data_GM12878.Rmd</code>) and HTML
(<code>docs/prepare_input_data_GM12878.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kevinlkx/footprint_clustering/blob/8e46a4664959487556b298cc0e8f7894c086c1cf/analysis/prepare_input_data_GM12878.Rmd" target="_blank">8e46a46</a>
</td>
<td>
kevinlkx
</td>
<td>
2025-09-10
</td>
<td>
wflow_publish("analysis/prepare_input_data_GM12878.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/kevinlkx/footprint_clustering/7d0b2e45413bb042fc2d7d660b3df928d7718556/docs/prepare_input_data_GM12878.html" target="_blank">7d0b2e4</a>
</td>
<td>
kevinlkx
</td>
<td>
2025-06-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kevinlkx/footprint_clustering/blob/715d7d8bb9f2ed68904d26744156e27c19e9ee87/analysis/prepare_input_data_GM12878.Rmd" target="_blank">715d7d8</a>
</td>
<td>
kevinlkx
</td>
<td>
2025-06-03
</td>
<td>
wflow_publish("analysis/prepare_*")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/kevinlkx/footprint_clustering/47e9251e433e3fb2dde8675948b202510955ec34/docs/prepare_input_data_GM12878.html" target="_blank">47e9251</a>
</td>
<td>
kevinlkx
</td>
<td>
2025-06-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/kevinlkx/footprint_clustering/blob/925f8bd231a84ea283017736138f3580b0f56ed0/analysis/prepare_input_data_GM12878.Rmd" target="_blank">925f8bd</a>
</td>
<td>
kevinlkx
</td>
<td>
2025-06-03
</td>
<td>
wflow_publish("analysis/prepare_input_data_GM12878.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Install <code>TOP</code> R package</p>
<pre class="r"><code># install.packages(&quot;devtools&quot;)
devtools::install_github(&quot;HarteminkLab/TOP&quot;)</code></pre>
<div id="input-data" class="section level1">
<h1>Input data</h1>
<p>Here, we show an example procedure with several steps for preparing
input data.</p>
<p>Load R packages</p>
<pre class="r"><code>library(TOP)
library(data.table)</code></pre>
<div id="step-1-find-tf-motif-matches-using-fimo-software"
class="section level2">
<h2>Step 1: Find TF motif matches using FIMO software</h2>
<p>To scan for TF motif matches,<br />
we use the <a href="https://meme-suite.org/meme/doc/fimo.html">FIMO</a>
software from <a href="https://meme-suite.org/meme/index.html">the MEME
suite</a>.</p>
<p>Download hg38 reference genome FASTA file and save it as
<code>hg38.fa</code>.</p>
<pre class="bash"><code>mkdir -p /project2/xinhe/kevinluo/footprint_clustering/data/ref_genome
cd /project2/xinhe/kevinluo/footprint_clustering/data/ref_genome
wget https://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/analysisSet/hg38.analysisSet.fa.gz
gunzip -c hg38.analysisSet.fa.gz &gt; hg38.fa</code></pre>
<p>Generate the <code>chrom.sizes</code> file which will be needed
later.</p>
<pre class="r"><code>index_fa(&#39;/project2/xinhe/kevinluo/footprint_clustering/data/ref_genome/hg38.fa&#39;, chromsize_file=&#39;/project2/xinhe/kevinluo/footprint_clustering/data/ref_genome/hg38.chrom.sizes&#39;)</code></pre>
<p>Download the motif files (in MEME format) from <a
href="https://jaspar.genereg.net/">JASPAR</a>.</p>
<pre class="bash"><code>mkdir -p /project2/xinhe/kevinluo/footprint_clustering/data/motifs/
cd /project2/xinhe/kevinluo/footprint_clustering/data/motifs/

wget https://jaspar.elixir.no/download/data/2024/CORE/JASPAR2024_CORE_non-redundant_pfms_meme.zip
unzip JASPAR2024_CORE_non-redundant_pfms_meme.zip -d JASPAR2024_CORE_non-redundant_pfms_meme</code></pre>
<p>Run FIMO</p>
</div>
<div id="step-2-get-candidate-tf-binding-sites" class="section level2">
<h2>Step 2: Get candidate TF binding sites</h2>
<p>We take motif matches obtained from FIMO as candidate binding sites,
and add 100 bp flanking regions on both sides of the motifs, then filter
candidate sites by FIMO p-value and PWM score, and filter the candidate
sites falling in ENCODE blacklist regions.</p>
<p>Download ENCODE blacklist from <a
href="https://www.encodeproject.org/annotations/ENCSR636HFF/">ENCODE
portal</a> and save as <code>blacklist.hg38.bed.gz</code>.</p>
<pre class="bash"><code>mkdir -p /project2/xinhe/kevinluo/footprint_clustering/data/ENCODE_blacklist
cd /project2/xinhe/kevinluo/footprint_clustering/data/ENCODE_blacklist
wget https://www.encodeproject.org/files/ENCFF356LFX/@@download/ENCFF356LFX.bed.gz
mv ENCFF356LFX.bed.gz blacklist.hg38.bed.gz</code></pre>
<p>Obtain the candidate sites</p>
<p>Using the following script to run step 1 and step 2 for different TF
motifs:</p>
<pre class="bash"><code>cp /project2/xinhe/kevinluo/footprint_clustering/data/motifs/JASPAR2024_CORE_non-redundant_pfms_meme/MA0139.2.meme /project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/

cp /project2/xinhe/kevinluo/footprint_clustering/data/motifs/JASPAR2024_CORE_non-redundant_pfms_meme/MA0138.3.meme /project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/

cp /project2/xinhe/kevinluo/footprint_clustering/data/motifs/JASPAR2024_CORE_non-redundant_pfms_meme/MA0506.1.meme /project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/

motif_dir=&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/&#39;

## CTCF
Rscript ~/projects/footprint_clustering/code/get_motif_sites.R \
  --tf=&#39;CTCF&#39; --motif=&#39;MA0139.2&#39; --motif_dir=$motif_dir \
  --threshP=1e-5 --threshPWM=10 --flank=100 \
  --outdir=&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/&#39;

## REST
Rscript ~/projects/footprint_clustering/code/get_motif_sites.R \
  --tf=&#39;REST&#39; --motif=&#39;MA0138.3&#39; --motif_dir=$motif_dir \
  --threshP=1e-5 --threshPWM=10 --flank=100 \
  --outdir=&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/&#39;

## NRF1
Rscript ~/projects/footprint_clustering/code/get_motif_sites.R \
  --tf=&#39;NRF1&#39; --motif=&#39;MA0506.1&#39; --motif_dir=$motif_dir \
  --threshP=1e-5 --threshPWM=10 --flank=100 \
  --outdir=&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/&#39;</code></pre>
</div>
<div id="step-3-count-dnase-seq-and-atc-seq-genome-wide-cleavage"
class="section level2">
<h2>Step 3: Count DNase-seq and ATC-seq genome-wide cleavage</h2>
<p>We use DNase-seq reads from GM12878 cell line (ENCODE ID:
<code>ENCSR000EMT</code>).</p>
<p>We first sort and index the BAM file, and obtain the total number of
mapped reads from the idxstats file, which will be used later when
normalizing read counts by library sizes.</p>
<pre class="bash"><code>module load samtools

mkdir -p /project2/xinhe/kevinluo/footprint_clustering/data/DNaseseq/GM12878
cd /project2/xinhe/kevinluo/footprint_clustering/data/DNaseseq/GM12878

# Download the BAM file from ENCODE
wget https://www.encodeproject.org/files/ENCFF020WZB/@@download/ENCFF020WZB.bam
wget https://www.encodeproject.org/files/ENCFF729UYK/@@download/ENCFF729UYK.bam

# Rename the bam file
mv ENCFF020WZB.bam DNaseseq_GM12878_alignments_rep1_ENCFF020WZB_hg38.bam
mv ENCFF729UYK.bam DNaseseq_GM12878_alignments_rep2_ENCFF729UYK_hg38.bam

samtools merge DNaseseq_GM12878_alignments_merged_hg38.bam DNaseseq_GM12878_alignments_rep1_ENCFF020WZB_hg38.bam DNaseseq_GM12878_alignments_rep2_ENCFF729UYK_hg38.bam

samtools sort DNaseseq_GM12878_alignments_merged_hg38.bam -o DNaseseq_GM12878_alignments_merged_sorted_hg38.bam

rm DNaseseq_GM12878_alignments_merged_hg38.bam</code></pre>
<pre class="r"><code># This BAM file has already been sorted, so we skip the sorting step. 
sort_index_idxstats_bam(&#39;/project2/xinhe/kevinluo/footprint_clustering/data/DNaseseq/GM12878/DNaseseq_GM12878_alignments_merged_sorted_hg38.bam&#39;, sort=FALSE, index=TRUE, idxstats=TRUE)</code></pre>
<pre class="r"><code>count_genome_cuts(bam_file=&#39;/project2/xinhe/kevinluo/footprint_clustering/data/DNaseseq/GM12878/DNaseseq_GM12878_alignments_merged_sorted_hg38.bam&#39;, 
                  chrom_size_file=&#39;/project2/xinhe/kevinluo/footprint_clustering/data/ref_genome/hg38.chrom.sizes&#39;, 
                  data_type=&#39;DNase&#39;,
                  outdir=&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/&#39;,
                  outname=&#39;GM12878.DNase&#39;)</code></pre>
<p>We use ATAC-seq reads from GM12878 cell line (ENCODE ID:
<code>ENCSR095QNB</code>) for example.</p>
<p>We first sort and index the BAM file, and obtain the total number of
mapped reads from the idxstats file, which will be used later when
normalizing read counts by library sizes.</p>
<pre class="bash"><code>module load samtools

mkdir -p /project2/xinhe/kevinluo/footprint_clustering/data/ATACseq/GM12878
cd /project2/xinhe/kevinluo/footprint_clustering/data/ATACseq/GM12878

# Download the BAM file from ENCODE
wget https://www.encodeproject.org/files/ENCFF415FEC/@@download/ENCFF415FEC.bam
wget https://www.encodeproject.org/files/ENCFF646NWY/@@download/ENCFF646NWY.bam

# Rename the bam file
mv ENCFF415FEC.bam ATACseq_GM12878_alignments_rep1_ENCFF415FEC_hg38.bam
mv ENCFF646NWY.bam ATACseq_GM12878_alignments_rep2_ENCFF646NWY_hg38.bam

samtools merge ATACseq_GM12878_alignments_merged_hg38.bam ATACseq_GM12878_alignments_rep1_ENCFF415FEC_hg38.bam ATACseq_GM12878_alignments_rep2_ENCFF646NWY_hg38.bam

samtools sort ATACseq_GM12878_alignments_merged_hg38.bam -o ATACseq_GM12878_alignments_merged_sorted_hg38.bam

rm ATACseq_GM12878_alignments_merged_hg38.bam</code></pre>
<pre class="r"><code># This BAM file has already been sorted, so we skip the sorting step. 
sort_index_idxstats_bam(&#39;/project/spott/kevinluo/Fiber_seq/data/ATACseq/ATACseq_GM12878_alignments_merged_hg38.bam&#39;, sort=FALSE, index=TRUE, idxstats=TRUE)</code></pre>
<p>Next, we count the ATAC reads along the genome, and save the genome
counts in BigWig files. This step may take a while especially for large
BAM files (the example BAM file we used here is very large, ~9.4 GB),
but it only needs to be done once. These BigWig files could be for
extracting the DNase or ATAC count matrices around the candidate sites
for different motifs.</p>
<p>For ATAC-seq, to address the offsets inherent in ATAC-seq reads, we
shift ATAC-seq read start positions by aligning the signal across
strands, thereby obtaining more accurate Tn5 binding locations
(Buenrostro et al., 2013).</p>
<pre class="r"><code>count_genome_cuts(bam_file=&#39;/project2/xinhe/kevinluo/footprint_clustering/data/ATACseq/GM12878/ATACseq_GM12878_alignments_merged_sorted_hg38.bam&#39;, 
                  chrom_size_file=&#39;/project2/xinhe/kevinluo/footprint_clustering/data/ref_genome/hg38.chrom.sizes&#39;, 
                  data_type=&#39;ATAC&#39;,
                  outdir=&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/&#39;,
                  outname=&#39;GM12878.ATAC&#39;)</code></pre>
</div>
<div
id="step-4-get-dnase--or-atac-seq-count-matrices-around-candidate-sites-then-normalize-bin-and-transform-the-counts"
class="section level2">
<h2>Step 4: Get DNase- or ATAC-seq count matrices around candidate
sites, then normalize, bin and transform the counts</h2>
<p>Get DNase-seq read counts matrix around candidate sites:</p>
<p>CTCF</p>
<pre class="r"><code>sites &lt;- readRDS(&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/CTCF_MA0139.2_1e-5.candidate.sites.rds&#39;)

count_matrix &lt;- get_sites_counts(sites,
                                 genomecount_dir=&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/&#39;,
                                 genomecount_name=&#39;GM12878.DNase&#39;)
saveRDS(count_matrix, &#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/CTCF.GM12878.DNase.counts.mat.rds&#39;)</code></pre>
<p>REST</p>
<pre class="r"><code>sites &lt;- readRDS(&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/REST_MA0138.3_1e-5.candidate.sites.rds&#39;)

count_matrix &lt;- get_sites_counts(sites,
                                 genomecount_dir=&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/&#39;,
                                 genomecount_name=&#39;GM12878.DNase&#39;)
saveRDS(count_matrix, &#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/REST.GM12878.DNase.counts.mat.rds&#39;)</code></pre>
<p>NRF1</p>
<pre class="r"><code>sites &lt;- readRDS(&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/NRF1_MA0506.1_1e-5.candidate.sites.rds&#39;)

count_matrix &lt;- get_sites_counts(sites,
                                 genomecount_dir=&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/&#39;,
                                 genomecount_name=&#39;GM12878.DNase&#39;)
saveRDS(count_matrix, &#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/NRF1.GM12878.DNase.counts.mat.rds&#39;)</code></pre>
<p>Get ATAC-seq read counts matrix around candidate sites:</p>
<p>CTCF</p>
<pre class="r"><code>sites &lt;- readRDS(&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/CTCF_MA0139.2_1e-5.candidate.sites.rds&#39;)

count_matrix &lt;- get_sites_counts(sites,
                                 genomecount_dir=&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/&#39;,
                                 genomecount_name=&#39;GM12878.ATAC&#39;)
saveRDS(count_matrix, &#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/CTCF.GM12878.ATAC.counts.mat.rds&#39;)</code></pre>
<p>REST</p>
<pre class="r"><code>sites &lt;- readRDS(&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/REST_MA0138.3_1e-5.candidate.sites.rds&#39;)

count_matrix &lt;- get_sites_counts(sites,
                                 genomecount_dir=&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/&#39;,
                                 genomecount_name=&#39;GM12878.ATAC&#39;)
saveRDS(count_matrix, &#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/REST.GM12878.ATAC.counts.mat.rds&#39;)</code></pre>
<p>NRF1</p>
<pre class="r"><code>sites &lt;- readRDS(&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/NRF1_MA0506.1_1e-5.candidate.sites.rds&#39;)

count_matrix &lt;- get_sites_counts(sites,
                                 genomecount_dir=&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/&#39;,
                                 genomecount_name=&#39;GM12878.ATAC&#39;)
saveRDS(count_matrix, &#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/NRF1.GM12878.ATAC.counts.mat.rds&#39;)</code></pre>
</div>
<div id="prepare-ctcf-chip-seq-data" class="section level2">
<h2>Prepare CTCF ChIP-seq data</h2>
<p>Download CTCF GM12878 ChIP-seq BAM files (ENCODE ID:
<code>ENCSR000DRZ</code>).</p>
<pre class="bash"><code>mkdir -p /project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/GM12878
cd /project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/GM12878

# Download the ChIP-seq BAM files
wget https://www.encodeproject.org/files/ENCFF430XCG/@@download/ENCFF430XCG.bam
wget https://www.encodeproject.org/files/ENCFF794BPW/@@download/ENCFF794BPW.bam

# Rename the BAM files
mv ENCFF430XCG.bam CTCF_GM12878_ChIPseq_rep1_ENCFF430XCG_hg38.bam
mv ENCFF794BPW.bam CTCF_GM12878_ChIPseq_rep2_ENCFF794BPW_hg38.bam</code></pre>
<p>Download CTCF ChIP-seq peaks</p>
<pre class="bash"><code>cd /project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/GM12878

# Download the ChIP-seq peaks
wget https://www.encodeproject.org/files/ENCFF951PEM/@@download/ENCFF951PEM.bed.gz

# Rename the peak files
mv ENCFF951PEM.bed.gz CTCF.GM12878.ChIPseq.peaks.bed.gz</code></pre>
<p>Sort and index the BAM files and obtain the number of mapped
reads.</p>
<pre class="r"><code># The BAM files have already been sorted, so we skip the sorting step. 
sort_index_idxstats_bam(&#39;/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/GM12878/CTCF_GM12878_ChIPseq_rep1_ENCFF430XCG_hg38.bam&#39;, sort=FALSE, index=TRUE, idxstats=TRUE)

sort_index_idxstats_bam(&#39;/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/GM12878/CTCF_GM12878_ChIPseq_rep2_ENCFF794BPW_hg38.bam&#39;, sort=FALSE, index=TRUE, idxstats=TRUE)</code></pre>
<p>Count ChIP-seq reads around candidate sites (merge ChIP-seq
replicates), and normalize to the reference ChIP-seq library size
(default: 20 million).</p>
<pre class="r"><code>sites &lt;- readRDS(&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/CTCF_MA0139.2_1e-5.candidate.sites.rds&#39;)

sites_chip &lt;- count_normalize_chip(sites,
                                   chip_bam_files=c(&#39;/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/GM12878/CTCF_GM12878_ChIPseq_rep1_ENCFF430XCG_hg38.bam&#39;,
                                                    &#39;/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/GM12878/CTCF_GM12878_ChIPseq_rep2_ENCFF794BPW_hg38.bam&#39;),
                                   chrom_size_file=&#39;/project2/xinhe/kevinluo/footprint_clustering/data/ref_genome/hg38.chrom.sizes&#39;)</code></pre>
<p>Add binary ChIP labels from ChIP-seq peaks</p>
<pre class="r"><code>sites_chip_labels &lt;- add_chip_peak_labels_to_sites(sites_chip,chip_peak_file=&#39;/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/GM12878/CTCF.GM12878.ChIPseq.peaks.bed.gz&#39;)

saveRDS(sites_chip_labels, &#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/CTCF.GM12878.sites.chip.labels.rds&#39;)</code></pre>
</div>
<div id="prepare-rest-chip-seq-data" class="section level2">
<h2>Prepare REST ChIP-seq data</h2>
<p>Download REST GM12878 ChIP-seq BAM files (ENCODE ID:
<code>ENCSR000BQS</code>).</p>
<pre class="bash"><code>mkdir -p /project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/GM12878
cd /project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/GM12878

# Download the ChIP-seq BAM files
wget https://www.encodeproject.org/files/ENCFF887ZNY/@@download/ENCFF887ZNY.bam
wget https://www.encodeproject.org/files/ENCFF213EBE/@@download/ENCFF213EBE.bam

# Rename the BAM files
mv ENCFF887ZNY.bam REST_GM12878_ChIPseq_rep1_ENCFF887ZNY_hg38.bam
mv ENCFF213EBE.bam REST_GM12878_ChIPseq_rep2_ENCFF213EBE_hg38.bam</code></pre>
<p>Download REST ChIP-seq peaks</p>
<pre class="bash"><code>cd /project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/GM12878

# Download the ChIP-seq peaks
wget https://www.encodeproject.org/files/ENCFF262MRD/@@download/ENCFF262MRD.bed.gz

# Rename the peak files
mv ENCFF262MRD.bed.gz REST.GM12878.ChIPseq.peaks.bed.gz</code></pre>
<p>Sort and index the BAM files and obtain the number of mapped
reads.</p>
<pre class="r"><code># The BAM files have already been sorted, so we skip the sorting step. 
sort_index_idxstats_bam(&#39;/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/GM12878/REST_GM12878_ChIPseq_rep1_ENCFF887ZNY_hg38.bam&#39;, sort=FALSE, index=TRUE, idxstats=TRUE)

sort_index_idxstats_bam(&#39;/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/GM12878/REST_GM12878_ChIPseq_rep2_ENCFF213EBE_hg38.bam&#39;, sort=FALSE, index=TRUE, idxstats=TRUE)</code></pre>
<p>Count ChIP-seq reads around candidate sites (merge ChIP-seq
replicates), and normalize to the reference ChIP-seq library size
(default: 20 million).</p>
<pre class="r"><code>sites &lt;- readRDS(&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/REST_MA0138.3_1e-5.candidate.sites.rds&#39;)

sites_chip &lt;- count_normalize_chip(sites,
                                   chip_bam_files=c(&#39;/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/GM12878/REST_GM12878_ChIPseq_rep1_ENCFF887ZNY_hg38.bam&#39;,
                                                    &#39;/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/GM12878/REST_GM12878_ChIPseq_rep2_ENCFF213EBE_hg38.bam&#39;),
                                   chrom_size_file=&#39;/project2/xinhe/kevinluo/footprint_clustering/data/ref_genome/hg38.chrom.sizes&#39;)</code></pre>
<p>Add binary ChIP labels from ChIP-seq peaksxw</p>
<pre class="r"><code>sites_chip_labels &lt;- add_chip_peak_labels_to_sites(sites_chip,chip_peak_file=&#39;/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/GM12878/REST.GM12878.ChIPseq.peaks.bed.gz&#39;)

saveRDS(sites_chip_labels, &#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/REST.GM12878.sites.chip.labels.rds&#39;)</code></pre>
</div>
<div id="prepare-nrf1-chip-seq-data" class="section level2">
<h2>Prepare NRF1 ChIP-seq data</h2>
<p>Download NRF1 GM12878 ChIP-seq BAM files (ENCODE ID:
<code>ENCSR000DZO</code>).</p>
<pre class="bash"><code>mkdir -p /project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/GM12878
cd /project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/GM12878

# Download the ChIP-seq BAM files
wget https://www.encodeproject.org/files/ENCFF588FQW/@@download/ENCFF588FQW.bam
wget https://www.encodeproject.org/files/ENCFF951GZL/@@download/ENCFF951GZL.bam

# Rename the BAM files
mv ENCFF588FQW.bam NRF1_GM12878_ChIPseq_rep1_ENCFF588FQW_hg38.bam
mv ENCFF951GZL.bam NRF1_GM12878_ChIPseq_rep2_ENCFF951GZL_hg38.bam</code></pre>
<p>Download NRF1 ChIP-seq peaks</p>
<pre class="bash"><code>cd /project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/GM12878

# Download the ChIP-seq peaks
wget https://www.encodeproject.org/files/ENCFF864EBU/@@download/ENCFF864EBU.bed.gz

# Rename the peak files
mv ENCFF864EBU.bed.gz NRF1.GM12878.ChIPseq.peaks.bed.gz</code></pre>
<p>Sort and index the BAM files and obtain the number of mapped
reads.</p>
<pre class="r"><code># The BAM files have already been sorted, so we skip the sorting step. 
sort_index_idxstats_bam(&#39;/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/GM12878/NRF1_GM12878_ChIPseq_rep1_ENCFF588FQW_hg38.bam&#39;, sort=FALSE, index=TRUE, idxstats=TRUE)

sort_index_idxstats_bam(&#39;/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/GM12878/NRF1_GM12878_ChIPseq_rep2_ENCFF951GZL_hg38.bam&#39;, sort=FALSE, index=TRUE, idxstats=TRUE)</code></pre>
<p>Count ChIP-seq reads around candidate sites (merge ChIP-seq
replicates), and normalize to the reference ChIP-seq library size
(default: 20 million).</p>
<pre class="r"><code>sites &lt;- readRDS(&#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/NRF1_MA0506.1_1e-5.candidate.sites.rds&#39;)

sites_chip &lt;- count_normalize_chip(sites,
                                   chip_bam_files=c(&#39;/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/GM12878/NRF1_GM12878_ChIPseq_rep1_ENCFF588FQW_hg38.bam&#39;,
                                                    &#39;/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/GM12878/NRF1_GM12878_ChIPseq_rep2_ENCFF951GZL_hg38.bam&#39;),
                                   chrom_size_file=&#39;/project2/xinhe/kevinluo/footprint_clustering/data/ref_genome/hg38.chrom.sizes&#39;)</code></pre>
<p>Add binary ChIP labels from ChIP-seq peaksxw</p>
<pre class="r"><code>sites_chip_labels &lt;- add_chip_peak_labels_to_sites(sites_chip,chip_peak_file=&#39;/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/GM12878/NRF1.GM12878.ChIPseq.peaks.bed.gz&#39;)

saveRDS(sites_chip_labels, &#39;/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/NRF1.GM12878.sites.chip.labels.rds&#39;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
