---
title: "Prepare input data in K562 cell line"
author: "Kaixuan Luo"
date: "`r Sys.Date()`"
output:
  workflowr::wflow_html:
    toc: true
editor_options:
  chunk_output_type: console
---

```{r, setup, include=FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  eval=FALSE,
  comment = "#>"
)
```

Install `TOP` R package
```{r, eval=FALSE}
# install.packages("devtools")
devtools::install_github("HarteminkLab/TOP")
```

Input data
==========

Here, we show an example procedure with several steps 
for preparing input data.

Load R packages
```{r load-packages, eval=FALSE, message=FALSE, warning=FALSE}
library(TOP)
library(data.table)
```

Step 1: Find TF motif matches using FIMO software
-------------------------------------------------

To scan for TF motif matches,  
we use the [FIMO][fimo] software from [the MEME suite][MEME].

Download hg38 reference genome FASTA file and save it as `hg38.fa`.
```{bash download-fasta, eval=FALSE}
mkdir -p /project2/xinhe/kevinluo/footprint_clustering/data/ref_genome
cd /project2/xinhe/kevinluo/footprint_clustering/data/ref_genome
wget https://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/analysisSet/hg38.analysisSet.fa.gz
gunzip -c hg38.analysisSet.fa.gz > hg38.fa
```

Generate the `chrom.sizes` file which will be needed later.
```{r, eval=FALSE}
index_fa('/project2/xinhe/kevinluo/footprint_clustering/data/ref_genome/hg38.fa', chromsize_file='/project2/xinhe/kevinluo/footprint_clustering/data/ref_genome/hg38.chrom.sizes')
```

Download the motif files (in MEME format) from [JASPAR][jaspar]. 
```{bash, eval=FALSE}
mkdir -p /project2/xinhe/kevinluo/footprint_clustering/data/motifs/
cd /project2/xinhe/kevinluo/footprint_clustering/data/motifs/

wget https://jaspar.elixir.no/download/data/2024/CORE/JASPAR2024_CORE_non-redundant_pfms_meme.zip
unzip JASPAR2024_CORE_non-redundant_pfms_meme.zip -d JASPAR2024_CORE_non-redundant_pfms_meme
```

Run FIMO

Step 2: Get candidate TF binding sites
--------------------------------------

We take motif matches obtained from FIMO 
as candidate binding sites, and add 100 bp flanking regions on 
both sides of the motifs, then filter candidate sites by FIMO p-value and PWM score,
and filter the candidate sites falling in ENCODE blacklist regions. 

Download ENCODE blacklist from [ENCODE portal](https://www.encodeproject.org/annotations/ENCSR636HFF/)
and save as `blacklist.hg38.bed.gz`.
```{bash, eval=FALSE}
mkdir -p /project2/xinhe/kevinluo/footprint_clustering/data/ENCODE_blacklist
cd /project2/xinhe/kevinluo/footprint_clustering/data/ENCODE_blacklist
wget https://www.encodeproject.org/files/ENCFF356LFX/@@download/ENCFF356LFX.bed.gz
mv ENCFF356LFX.bed.gz blacklist.hg38.bed.gz
```

Obtain the candidate sites

Using the following script to run step 1 and step 2 for different TF motifs: 
```{bash, eval=FALSE}
cp /project2/xinhe/kevinluo/footprint_clustering/data/motifs/JASPAR2024_CORE_non-redundant_pfms_meme/MA0139.2.meme /project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/

cp /project2/xinhe/kevinluo/footprint_clustering/data/motifs/JASPAR2024_CORE_non-redundant_pfms_meme/MA0138.3.meme /project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/

cp /project2/xinhe/kevinluo/footprint_clustering/data/motifs/JASPAR2024_CORE_non-redundant_pfms_meme/MA0506.1.meme /project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/

motif_dir='/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/'

## CTCF
Rscript ~/projects/footprint_clustering/code/get_motif_sites.R \
  --tf='CTCF' --motif='MA0139.2' --motif_dir=$motif_dir \
  --threshP=1e-5 --threshPWM=10 --flank=100 \
  --outdir='/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/'

## REST
Rscript ~/projects/footprint_clustering/code/get_motif_sites.R \
  --tf='REST' --motif='MA0138.3' --motif_dir=$motif_dir \
  --threshP=1e-5 --threshPWM=10 --flank=100 \
  --outdir='/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/'

## NRF1
Rscript ~/projects/footprint_clustering/code/get_motif_sites.R \
  --tf='NRF1' --motif='MA0506.1' --motif_dir=$motif_dir \
  --threshP=1e-5 --threshPWM=10 --flank=100 \
  --outdir='/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/'
```

Step 3: Count DNase-seq and ATC-seq genome-wide cleavage
-----------------------------------------------------

### DNase-seq data

We use DNase-seq reads from K562 cell line (ENCODE ID: `ENCSR000EMT`). 

We first sort and index the BAM file, 
and obtain the total number of mapped reads from the idxstats file,
which will be used later 
when normalizing read counts by library sizes.

```{bash, eval=FALSE}
module load samtools

mkdir -p /project2/xinhe/kevinluo/footprint_clustering/data/DNaseseq/K562
cd /project2/xinhe/kevinluo/footprint_clustering/data/DNaseseq/K562

# Download the BAM file from ENCODE
wget https://www.encodeproject.org/files/ENCFF257HEE/@@download/ENCFF257HEE.bam

# Sort the bam file
samtools sort ENCFF257HEE.bam -o DNaseseq_K562_alignments_sorted_hg38.bam

rm ENCFF257HEE.bam
```

```{r, eval=FALSE, message=FALSE, warning=FALSE}
# This BAM file has already been sorted, so we skip the sorting step. 
sort_index_idxstats_bam('/project2/xinhe/kevinluo/footprint_clustering/data/DNaseseq/K562/DNaseseq_K562_alignments_sorted_hg38.bam', sort=FALSE, index=TRUE, idxstats=TRUE)
```

```{r, eval=FALSE}
count_genome_cuts(bam_file='/project2/xinhe/kevinluo/footprint_clustering/data/DNaseseq/K562/DNaseseq_K562_alignments_sorted_hg38.bam', 
                  chrom_size_file='/project2/xinhe/kevinluo/footprint_clustering/data/ref_genome/hg38.chrom.sizes', 
                  data_type='DNase',
                  outdir='/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/',
                  outname='K562.DNase')
```

### ATAC-seq data

We use ATAC-seq reads from K562 cell line (ENCODE ID: `ENCSR483RKN`) for example. 

We first sort and index the BAM file, 
and obtain the total number of mapped reads from the idxstats file,
which will be used later 
when normalizing read counts by library sizes.

```{bash, eval=FALSE}
module load samtools

mkdir -p /project2/xinhe/kevinluo/footprint_clustering/data/ATACseq/K562
cd /project2/xinhe/kevinluo/footprint_clustering/data/ATACseq/K562

# Download the BAM file from ENCODE
wget https://www.encodeproject.org/files/ENCFF512VEZ/@@download/ENCFF512VEZ.bam
wget https://www.encodeproject.org/files/ENCFF987XOV/@@download/ENCFF987XOV.bam

# Rename the bam file
mv ENCFF512VEZ.bam ATACseq_K562_alignments_rep1_ENCFF512VEZ_hg38.bam
mv ENCFF987XOV.bam ATACseq_K562_alignments_rep2_ENCFF987XOV_hg38.bam

samtools merge ATACseq_K562_alignments_merged_hg38.bam ATACseq_K562_alignments_rep1_ENCFF512VEZ_hg38.bam ATACseq_K562_alignments_rep2_ENCFF987XOV_hg38.bam

samtools sort ATACseq_K562_alignments_merged_hg38.bam -o ATACseq_K562_alignments_merged_sorted_hg38.bam

rm ATACseq_K562_alignments_merged_hg38.bam
rm ATACseq_K562_alignments_rep1_ENCFF512VEZ_hg38.bam
rm ATACseq_K562_alignments_rep2_ENCFF987XOV_hg38.bam
```

```{r, eval=FALSE, message=FALSE, warning=FALSE}
# This BAM file has already been sorted, so we skip the sorting step. 
sort_index_idxstats_bam('/project2/xinhe/kevinluo/footprint_clustering/data/ATACseq/K562/ATACseq_K562_alignments_merged_sorted_hg38.bam', sort=FALSE, index=TRUE, idxstats=TRUE)
```

```{r, eval=FALSE}
count_genome_cuts(bam_file='/project2/xinhe/kevinluo/footprint_clustering/data/ATACseq/K562/ATACseq_K562_alignments_merged_sorted_hg38.bam', 
                  chrom_size_file='/project2/xinhe/kevinluo/footprint_clustering/data/ref_genome/hg38.chrom.sizes', 
                  data_type='ATAC',
                  outdir='/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/',
                  outname='K562.ATAC')
```


Step 4: Get DNase- or ATAC-seq count matrices around candidate sites, then normalize, bin and transform the counts
------------------------------------------------------------------------------------------------------------------

### Get DNase-seq read counts matrix around candidate sites:

CTCF
```{r, eval=FALSE}
sites <- readRDS('/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/CTCF_MA0139.2_1e-5.candidate.sites.rds')

count_matrix <- get_sites_counts(sites,
                                 genomecount_dir='/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/',
                                 genomecount_name='K562.DNase')
saveRDS(count_matrix, '/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/CTCF.K562.DNase.counts.mat.rds')
```

REST
```{r, eval=FALSE}
sites <- readRDS('/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/REST_MA0138.3_1e-5.candidate.sites.rds')

count_matrix <- get_sites_counts(sites,
                                 genomecount_dir='/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/',
                                 genomecount_name='K562.DNase')
saveRDS(count_matrix, '/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/REST.K562.DNase.counts.mat.rds')
```

NRF1
```{r, eval=FALSE}
sites <- readRDS('/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/NRF1_MA0506.1_1e-5.candidate.sites.rds')

count_matrix <- get_sites_counts(sites,
                                 genomecount_dir='/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/',
                                 genomecount_name='K562.DNase')
saveRDS(count_matrix, '/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/NRF1.K562.DNase.counts.mat.rds')
```


### Get ATAC-seq read counts matrix around candidate sites:

CTCF
```{r, eval=FALSE}
sites <- readRDS('/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/CTCF_MA0139.2_1e-5.candidate.sites.rds')

count_matrix <- get_sites_counts(sites,
                                 genomecount_dir='/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/',
                                 genomecount_name='K562.ATAC')
saveRDS(count_matrix, '/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/CTCF.K562.ATAC.counts.mat.rds')
```

CTCF (ENCSR483RKN)
```{r, eval=FALSE}
sites <- readRDS('/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/CTCF_MA0139.2_1e-5.candidate.sites.rds')

count_matrix <- get_sites_counts(sites,
                                 genomecount_dir='/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/',
                                 genomecount_name='K562.ATAC.ENCSR483RKN')
saveRDS(count_matrix, '/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/CTCF.K562.ATAC.ENCSR483RKN.counts.mat.rds')
```

REST
```{r, eval=FALSE}
sites <- readRDS('/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/REST_MA0138.3_1e-5.candidate.sites.rds')

count_matrix <- get_sites_counts(sites,
                                 genomecount_dir='/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/',
                                 genomecount_name='K562.ATAC')
saveRDS(count_matrix, '/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/REST.K562.ATAC.counts.mat.rds')
```

NRF1
```{r, eval=FALSE}
sites <- readRDS('/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/NRF1_MA0506.1_1e-5.candidate.sites.rds')

count_matrix <- get_sites_counts(sites,
                                 genomecount_dir='/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/',
                                 genomecount_name='K562.ATAC')
saveRDS(count_matrix, '/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/NRF1.K562.ATAC.counts.mat.rds')
```

Prepare CTCF ChIP-seq data
---------------------------------------------------------------------

### CTCF ChIP-seq from ENCSR000EGM (Stanford)

Download CTCF K562 ChIP-seq BAM files (ENCODE ID: `ENCSR000EGM`).

```{bash, eval=FALSE}
mkdir -p /project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562
cd /project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562

# Download the ChIP-seq BAM files
wget https://www.encodeproject.org/files/ENCFF172KOJ/@@download/ENCFF172KOJ.bam
wget https://www.encodeproject.org/files/ENCFF265ZSP/@@download/ENCFF265ZSP.bam

# Rename the BAM files
mv ENCFF172KOJ.bam CTCF_K562_ChIPseq_rep1_ENCFF430XCG_hg38.bam
mv ENCFF265ZSP.bam CTCF_K562_ChIPseq_rep2_ENCFF794BPW_hg38.bam
```

Download CTCF ChIP-seq peaks from 

```{bash, eval=FALSE}
cd /project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562

# Download the ChIP-seq peaks
wget https://www.encodeproject.org/files/ENCFF660GHM/@@download/ENCFF660GHM.bed.gz

# Rename the peak files
mv ENCFF660GHM.bed.gz CTCF.K562.ChIPseq.peaks.bed.gz
```

Sort and index the BAM files and obtain the number of mapped reads.
```{r, eval=FALSE}
# The BAM files have already been sorted, so we skip the sorting step. 
sort_index_idxstats_bam('/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562/CTCF_K562_ChIPseq_rep1_ENCFF430XCG_hg38.bam', sort=FALSE, index=TRUE, idxstats=TRUE)

sort_index_idxstats_bam('/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562/CTCF_K562_ChIPseq_rep2_ENCFF794BPW_hg38.bam', sort=FALSE, index=TRUE, idxstats=TRUE)
```

Count ChIP-seq reads around candidate sites (merge ChIP-seq replicates),
and normalize to the reference ChIP-seq library size (default: 20 million).
```{r, eval=FALSE}
sites <- readRDS('/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/CTCF_MA0139.2_1e-5.candidate.sites.rds')

sites_chip <- count_normalize_chip(sites,
chip_bam_files=c('/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562/CTCF_K562_ChIPseq_rep1_ENCFF430XCG_hg38.bam',
'/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562/CTCF_K562_ChIPseq_rep2_ENCFF794BPW_hg38.bam'),
                                   chrom_size_file='/project2/xinhe/kevinluo/footprint_clustering/data/ref_genome/hg38.chrom.sizes')
```

Add binary ChIP labels from ChIP-seq peaks
```{r eval=FALSE}
sites_chip_labels <- add_chip_peak_labels_to_sites(sites_chip,chip_peak_file='/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562/CTCF.K562.ChIPseq.peaks.bed.gz')

saveRDS(sites_chip_labels, '/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/CTCF.K562.sites.chip.labels.rds')
```


### CTCF ChIP-seq from ENCSR000DWE (UW)

Download CTCF K562 ChIP-seq BAM files (ENCODE ID: `ENCSR000DWE`).

```{bash, eval=FALSE}
cd /project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562

# Download the ChIP-seq BAM files
wget https://www.encodeproject.org/files/ENCFF800GVR/@@download/ENCFF800GVR.bam
wget https://www.encodeproject.org/files/ENCFF196QVZ/@@download/ENCFF196QVZ.bam

# Rename the BAM files
mv ENCFF800GVR.bam CTCF_K562_ChIPseq_UW_rep1_ENCFF800GVR_hg38.bam
mv ENCFF196QVZ.bam CTCF_K562_ChIPseq_UW_rep2_ENCFF196QVZ_hg38.bam
```

Download CTCF ChIP-seq peaks from 

```{bash, eval=FALSE}
cd /project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562

# Download the ChIP-seq peaks
wget https://www.encodeproject.org/files/ENCFF736NYC/@@download/ENCFF736NYC.bed.gz

# Rename the peak files
mv ENCFF736NYC.bed.gz CTCF.K562.ChIPseq.UW.peaks.bed.gz
```

Sort and index the BAM files and obtain the number of mapped reads.
```{r, eval=FALSE}
# The BAM files have already been sorted, so we skip the sorting step. 
sort_index_idxstats_bam('/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562/CTCF_K562_ChIPseq_UW_rep1_ENCFF800GVR_hg38.bam', sort=FALSE, index=TRUE, idxstats=TRUE)

sort_index_idxstats_bam('/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562/CTCF_K562_ChIPseq_UW_rep2_ENCFF196QVZ_hg38.bam', sort=FALSE, index=TRUE, idxstats=TRUE)
```

Count ChIP-seq reads around candidate sites (merge ChIP-seq replicates),
and normalize to the reference ChIP-seq library size (default: 20 million).
```{r, eval=FALSE}
sites <- readRDS('/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/CTCF_MA0139.2_1e-5.candidate.sites.rds')

sites_chip <- count_normalize_chip(sites,
                                   chip_bam_files=c('/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562/CTCF_K562_ChIPseq_UW_rep1_ENCFF800GVR_hg38.bam',
                                                    '/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562/CTCF_K562_ChIPseq_UW_rep2_ENCFF196QVZ_hg38.bam'),
                                   chrom_size_file='/project2/xinhe/kevinluo/footprint_clustering/data/ref_genome/hg38.chrom.sizes')
```

Add binary ChIP labels from ChIP-seq peaks
```{r eval=FALSE}
sites_chip_labels <- add_chip_peak_labels_to_sites(sites_chip,chip_peak_file='/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562/CTCF.K562.ChIPseq.UW.peaks.bed.gz')

saveRDS(sites_chip_labels, '/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/CTCF.K562.UW.sites.chip.labels.rds')
```


### CTCF ChIP-seq from ENCSR000AKO (Broad)

Download CTCF K562 ChIP-seq BAM files (ENCODE ID: `ENCSR000AKO`).

```{bash, eval=FALSE}
cd /project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562

# Download the ChIP-seq BAM files
wget https://www.encodeproject.org/files/ENCFF834MPG/@@download/ENCFF834MPG.bam
wget https://www.encodeproject.org/files/ENCFF173QSV/@@download/ENCFF173QSV.bam

# Rename the BAM files
mv ENCFF834MPG.bam CTCF_K562_ChIPseq_Broad_rep1_ENCFF834MPG_hg38.bam
mv ENCFF173QSV.bam CTCF_K562_ChIPseq_Broad_rep2_ENCFF173QSV_hg38.bam
```

Download CTCF ChIP-seq peaks from 

```{bash, eval=FALSE}
cd /project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562

# Download the ChIP-seq peaks
wget https://www.encodeproject.org/files/ENCFF769AUF/@@download/ENCFF769AUF.bed.gz

# Rename the peak files
mv ENCFF769AUF.bed.gz CTCF.K562.ChIPseq.Broad.peaks.bed.gz
```

Sort and index the BAM files and obtain the number of mapped reads.
```{r, eval=FALSE}
# The BAM files have already been sorted, so we skip the sorting step. 
sort_index_idxstats_bam('/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562/CTCF_K562_ChIPseq_Broad_rep1_ENCFF834MPG_hg38.bam', sort=FALSE, index=TRUE, idxstats=TRUE)

sort_index_idxstats_bam('/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562/CTCF_K562_ChIPseq_Broad_rep2_ENCFF173QSV_hg38.bam', sort=FALSE, index=TRUE, idxstats=TRUE)
```

Count ChIP-seq reads around candidate sites (merge ChIP-seq replicates),
and normalize to the reference ChIP-seq library size (default: 20 million).
```{r, eval=FALSE}
sites <- readRDS('/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/CTCF_MA0139.2_1e-5.candidate.sites.rds')

sites_chip <- count_normalize_chip(sites,
                                   chip_bam_files=c('/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562/CTCF_K562_ChIPseq_Broad_rep1_ENCFF834MPG_hg38.bam',
                                                    '/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562/CTCF_K562_ChIPseq_Broad_rep2_ENCFF173QSV_hg38.bam'),
                                   chrom_size_file='/project2/xinhe/kevinluo/footprint_clustering/data/ref_genome/hg38.chrom.sizes')
```

Add binary ChIP labels from ChIP-seq peaks
```{r eval=FALSE}
sites_chip_labels <- add_chip_peak_labels_to_sites(sites_chip,chip_peak_file='/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562/CTCF.K562.ChIPseq.Broad.peaks.bed.gz')

saveRDS(sites_chip_labels, '/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/CTCF.K562.Broad.sites.chip.labels.rds')
```

Prepare REST ChIP-seq data
---------------------------------------------------------------------

Download REST K562 ChIP-seq BAM files (ENCODE ID: `ENCSR137ZMQ`).

```{bash, eval=FALSE}
mkdir -p /project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562
cd /project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562

# Download the ChIP-seq BAM files
wget https://www.encodeproject.org/files/ENCFF116CTI/@@download/ENCFF116CTI.bam
wget https://www.encodeproject.org/files/ENCFF778MNM/@@download/ENCFF778MNM.bam

# Rename the BAM files
mv ENCFF116CTI.bam REST_K562_ChIPseq_rep1_ENCFF430XCG_hg38.bam
mv ENCFF778MNM.bam REST_K562_ChIPseq_rep2_ENCFF794BPW_hg38.bam
```

Download REST ChIP-seq peaks

```{bash, eval=FALSE}
cd /project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562

# Download the ChIP-seq peaks
wget https://www.encodeproject.org/files/ENCFF761YYL/@@download/ENCFF761YYL.bed.gz

# Rename the peak files
mv ENCFF761YYL.bed.gz REST.K562.ChIPseq.peaks.bed.gz
```

Sort and index the BAM files and obtain the number of mapped reads.
```{r, eval=FALSE}
# The BAM files have already been sorted, so we skip the sorting step. 
sort_index_idxstats_bam('/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562/REST_K562_ChIPseq_rep1_ENCFF430XCG_hg38.bam', sort=FALSE, index=TRUE, idxstats=TRUE)

sort_index_idxstats_bam('/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562/REST_K562_ChIPseq_rep2_ENCFF794BPW_hg38.bam', sort=FALSE, index=TRUE, idxstats=TRUE)
```

Count ChIP-seq reads around candidate sites (merge ChIP-seq replicates),
and normalize to the reference ChIP-seq library size (default: 20 million).
```{r, eval=FALSE}
sites <- readRDS('/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/REST_MA0138.3_1e-5.candidate.sites.rds')

sites_chip <- count_normalize_chip(sites,
chip_bam_files=c('/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562/REST_K562_ChIPseq_rep1_ENCFF430XCG_hg38.bam',
'/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562/REST_K562_ChIPseq_rep2_ENCFF794BPW_hg38.bam'),
                                   chrom_size_file='/project2/xinhe/kevinluo/footprint_clustering/data/ref_genome/hg38.chrom.sizes')
```

Add binary ChIP labels from ChIP-seq peaks

```{r eval=FALSE}
sites_chip_labels <- add_chip_peak_labels_to_sites(sites_chip,chip_peak_file='/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562/REST.K562.ChIPseq.peaks.bed.gz')

saveRDS(sites_chip_labels, '/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/REST.K562.sites.chip.labels.rds')
```


Prepare NRF1 ChIP-seq data
---------------------------------------------------------------------

Download NRF1 K562 ChIP-seq BAM files (ENCODE ID: `ENCSR837EYC`).

```{bash, eval=FALSE}
mkdir -p /project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562
cd /project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562

# Download the ChIP-seq BAM files
wget https://www.encodeproject.org/files/ENCFF265VZX/@@download/ENCFF265VZX.bam
wget https://www.encodeproject.org/files/ENCFF677KJK/@@download/ENCFF677KJK.bam

# Rename the BAM files
mv ENCFF265VZX.bam NRF1_K562_ChIPseq_rep1_ENCFF265VZX_hg38.bam
mv ENCFF677KJK.bam NRF1_K562_ChIPseq_rep2_ENCFF677KJK_hg38.bam
```

Download NRF1 ChIP-seq peaks

```{bash, eval=FALSE}
cd /project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562

# Download the ChIP-seq peaks
wget https://www.encodeproject.org/files/ENCFF259YUE/@@download/ENCFF259YUE.bed.gz

# Rename the peak files
mv ENCFF259YUE.bed.gz NRF1.K562.ChIPseq.peaks.bed.gz
```

Sort and index the BAM files and obtain the number of mapped reads.
```{r, eval=FALSE}
# The BAM files have already been sorted, so we skip the sorting step. 
sort_index_idxstats_bam('/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562/NRF1_K562_ChIPseq_rep1_ENCFF265VZX_hg38.bam', sort=FALSE, index=TRUE, idxstats=TRUE)

sort_index_idxstats_bam('/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562/NRF1_K562_ChIPseq_rep2_ENCFF677KJK_hg38.bam', sort=FALSE, index=TRUE, idxstats=TRUE)
```

Count ChIP-seq reads around candidate sites (merge ChIP-seq replicates),
and normalize to the reference ChIP-seq library size (default: 20 million).
```{r, eval=FALSE}
sites <- readRDS('/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/NRF1_MA0506.1_1e-5.candidate.sites.rds')

sites_chip <- count_normalize_chip(sites,
chip_bam_files=c('/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562/NRF1_K562_ChIPseq_rep1_ENCFF265VZX_hg38.bam',
'/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562/NRF1_K562_ChIPseq_rep2_ENCFF677KJK_hg38.bam'),
                                   chrom_size_file='/project2/xinhe/kevinluo/footprint_clustering/data/ref_genome/hg38.chrom.sizes')
```

Add binary ChIP labels from ChIP-seq peaks
```{r eval=FALSE}
sites_chip_labels <- add_chip_peak_labels_to_sites(sites_chip,chip_peak_file='/project2/xinhe/kevinluo/footprint_clustering/data/ChIPseq/K562/NRF1.K562.ChIPseq.peaks.bed.gz')

saveRDS(sites_chip_labels, '/project2/xinhe/kevinluo/footprint_clustering/processed_data/hg38/NRF1.K562.sites.chip.labels.rds')
```


[jaspar]: https://jaspar.genereg.net/
[Snakemake]: https://snakemake.readthedocs.io/en/stable/
[top-pipeline]: https://github.com/HarteminkLab/TOP-paper-resources/tree/main/code/snakemake
[top-resources-website]: https://harteminklab.github.io/TOP-paper-resources
[bedtools]: https://bedtools.readthedocs.io/en/latest/
[bwtool]: https://github.com/CRG-Barcelona/bwtool/wiki
[GenomicRanges]: https://bioconductor.org/packages/release/bioc/html/GenomicRanges.html
[Rsamtools]: https://bioconductor.org/packages/release/bioc/html/Rsamtools.html
[data.table]: https://cran.r-project.org/web/packages/data.table/index.html
[doParallel]: https://cran.r-project.org/web/packages/doParallel/index.html
[ggplot2]: https://ggplot2.tidyverse.org
[fimo]: https://meme-suite.org/meme/doc/fimo.html
[bedGraphToBigWig]:https://genome.ucsc.edu/goldenpath/help/bigWig.html
[bigWigAverageOverBed]:https://genome.ucsc.edu/goldenpath/help/bigWig.html
[UCSC utilities]: http://hgdownload.soe.ucsc.edu/admin/exe/
[MEME]:https://meme-suite.org/meme/index.html
